suite-native/module-device-onboarding/src/components/SwipeableWalkthrough/SwipeableWalkthrough.tsx

```diff
@@ -0,0 +1,51 @@
+import { ReactNode, useState } from 'react';
+import { LayoutChangeEvent } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import { runOnJS } from 'react-native-reanimated';
+
+import { Box } from '@suite-native/atoms';
+import { getScreenHeight } from '@trezor/env-utils';
+
+type SwipeableWalkthroughProps = {
+    children: ReactNode;
+    onSwipeUp: () => void;
+    onSwipeDown: () => void;
+};
+
+const PAN_GESTURE_DETECTION_THRESHOLD = 50;
+
+export const SwipeableWalkthrough = ({
+    children,
+    onSwipeUp,
+    onSwipeDown,
+}: SwipeableWalkthroughProps) => {
+    const [isPanGestureEnabled, setIsPanGestureEnabled] = useState(false);
+
+    const handleContentSizeChange = (event: LayoutChangeEvent) => {
+        const screenHeight = getScreenHeight();
+        const { height } = event.nativeEvent.layout;
+
+        //pan gesture is disabled if the content is scrollable
+        setIsPanGestureEnabled(height <= screenHeight);
+    };
+
+    const panGesture = Gesture.Pan()
+        .enabled(isPanGestureEnabled)
+        .onEnd(event => {
+            const { translationY } = event;
+
+            if (translationY < -PAN_GESTURE_DETECTION_THRESHOLD) {
+                runOnJS(onSwipeUp)();
+            } else if (translationY > PAN_GESTURE_DETECTION_THRESHOLD) {
+                runOnJS(onSwipeDown)();
+            }
+        });
+
+    return (
+        <GestureDetector gesture={panGesture}>
+            <Box flex={1} onLayout={handleContentSizeChange}>
```

Comment:
Looking forward to being able to do this with the new architecture instead :D  https://reactnative.dev/blog/2024/10/23/the-new-architecture-is-here#uselayouteffect